source/common/http/conn_manager_impl.cc

```diff
@@ -920,6 +934,20 @@ void ConnectionManagerImpl::ActiveStream::callLowWatermarkCallbacks() {
   }
 }
 
+void ConnectionManagerImpl::ActiveStream::setBufferLimit(uint32_t limit) {
+  uint32_t new_limit = morePermissiveLimit(limit, buffer_limit_);
```

Comment:
do you mean to assign new_limit to buffer_limit_ below? (also nit: const on new_limit)